File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
class Solution {
    public int[] solution(int[] arr, int n) {
        int[] answer = new int[arr.length];
        boolean flag = arr.length%2==0? true:false;
        if(flag) {
            for(int i=0; i<arr.length; i++) {
                if(i%2!=0)
                    answer[i] = arr[i] + n;
                else answer[i] = arr[i];
            }
        } else {
            for(int i=0; i<arr.length; i++) {
                if(i%2==0)
                    answer[i] = arr[i] + n;
                else answer[i] = arr[i];
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, int n) {
    vector<int> answer;
    for(int i=0; i<arr.size(); i++) {
        if (arr.size()%2 == 0) {
            if (i%2!=0) answer.push_back(arr[i]+n);
            else answer.push_back(arr[i]);
        } else {
            if (i%2==0) answer.push_back(arr[i]+n);
            else answer.push_back(arr[i]);
        }

    }
    return answer;
}